### ISSUE TITLE
Reward System

### ISSUE THREAD

**Issue by snomann72** [conan513/SingleCore_TC#60]
In the previous updates, the GM Bot rewarded a player a gift for achieving 10 levels.  The latest beta update 5, that GM Bot, which was GM Happypants, is MIA and not rewarding that gift.  It seems the latest beta update 5 has broken the reward system.

**Comment by conan513** [comment]
I think it's fixed in up5beta2

**Comment by snomann72** [comment]
Conan, just tested the Reward system with the latest patch.  It is still broken since the Early Beta Release 5 and Beta 5 Update 2.  I did notice that it had a white text saying that the quest system is skipped when patching the Arch-SPP.  Also seen it in the worldserver log stating that it skipped it.

Something about is showing 999999 or something to that.  Here is what the log is showing.

Quest 190010 has `RewardChoiceItemId1` = 999999 but item with entry 999999 does not exist, quest will not reward this item.
Table `creature_queststarter` has data for nonexistent creature entry (190004) and existed quest 190010
Table `creature_questender` has data for nonexistent creature entry (190004) and existed quest 190010

**Comment by thesawolf** [comment]
The 10 levels achievement rewards are handled setup via the 4 SQL files titled QOL-world-XXXXXXXX.sql in: https://github.com/thesawolf/TrinityCore/tree/TrinityCoreLegacy/sql/TrinityCore-Patches/LegacyTrinityCore

https://github.com/thesawolf/TrinityCore/blob/TrinityCoreLegacy/sql/TrinityCore-Patches/LegacyTrinityCore/QOL-world-achievement_reward.sql   handles setting up the achievements with an appropriate mail set every 10 levels (as well as fishing and cooking rewards)

https://github.com/thesawolf/TrinityCore/blob/TrinityCoreLegacy/sql/TrinityCore-Patches/LegacyTrinityCore/QOL-world-creature_template.sql  adds GM HappyPants as a valid mail sender

https://github.com/thesawolf/TrinityCore/blob/TrinityCoreLegacy/sql/TrinityCore-Patches/LegacyTrinityCore/QOL-world-item_loot_template.sql  handles the actual loot rewards in the achievement mails by making the random loot chances in the various gift boxes awarded

https://github.com/thesawolf/TrinityCore/blob/TrinityCoreLegacy/sql/TrinityCore-Patches/LegacyTrinityCore/QOL-world-item_template.sql  changes Landro's boxes around so there's one Pet one, one Mount one and one Gift one (by putting in random options for ALL the various special loot items that you could only get from the collectible card game and collector's editions, etc.)

**Comment by snomann72** [comment]
Thesawolf, I added the sql files and nothing broken in the SQL.  Found 0 rows and 0 warnings.  I made a test toon, and he hit level 10 earlier.  However, there is no mail for him to get when he hit level 10.

I am thinking GM HappyPants is missing in this beta update.

Going to retest in the morning to see if it has resolved.  Looked at a fresh log and the creater_queststarter and creature-questender is not there now.  Will give a update if he shows up.
